.github/workflows/deploy.yml

```
@@ -0,0 +1,87 @@
+name: Deployment
+
+on:
+  workflow_call:
+    inputs:
+      python_version:
+        required: true
+        type: string
+      environment_id:
+        required: true
+        type: string
+      environment_suffix:
+        required: true
+        type: string
+      dynamic_cf_parameters:
+        required: true
+        type: string
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Checkout
+        id: checkout
+        uses: actions/checkout@v3
+
+      - name: Set up Python ${{ inputs.python_version }}
+        id: python-setup
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ inputs.python_version }}
```

I'd suggest that we make the following change:
```suggestion
          python-version: "${{ inputs.python_version }}"
```
because, if left without quotes, yaml will parse something like `3.10` as a float and turn it to `3.1` which will break this step.

---

It is not an issue as the workflow input is a string
```
python_version:
  required: true
  type: string
````
So, we can't give 3.10 as an input, it will error out. We can only give "3.10" which will be propagated as string down the line. Still, I will change it for consistency.

---

Addressed in cd181227faf020419923b41d9c50aa85683ae065